apps/dashboard/components/logs/chart/utils/format-timestamp.ts

```diff
@@ -1,9 +1,38 @@
 import type { CompoundTimeseriesGranularity } from "@/lib/trpc/routers/utils/granularity";
 import { format, fromUnixTime } from "date-fns";
 
+// Memoization cache with bounded size
+const formatCache = new Map<string, string>();
+const MAX_CACHE_SIZE = 1000;
+
+// Memoized format function
```

Comment:
Can we add a small comment about why we are doing this like motivation type of explanation.